File: Habitica/src/main/java/com/habitrpg/android/habitica/models/invitations/Invitations.java
package com.habitrpg.android.habitica.models.invitations;

import com.habitrpg.android.habitica.models.user.User;

import java.util.List;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Invitations extends RealmObject {

    @PrimaryKey
    private String userId;

    User user;
    private PartyInvite party;
    private RealmList<GuildInvite> guilds;

    /**
     * @return The party invite
     */
    public PartyInvite getParty() {
        return party;
    }

    /**
     * @param party The party
     */
    public void setParty(PartyInvite party) {
        this.party = party;
    }

    /**
     * @return The guilds invite
     */
    public List<GuildInvite> getGuilds() {
        return guilds;
    }

    /**
     * @param guilds The guilds
     */
    public void setGuilds(RealmList<GuildInvite> guilds) {
        this.guilds = guilds;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/invitations/Invitations.kt
package com.habitrpg.android.habitica.models.invitations

import com.habitrpg.android.habitica.models.BaseObject
import com.habitrpg.android.habitica.models.user.User

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass

@RealmClass(embedded = true)
open class Invitations : RealmObject(), BaseObject {
    var party: PartyInvite? = null
    var parties: RealmList<PartyInvite>? = null
    var guilds: RealmList<GuildInvite>? = null

    fun removeInvitation(groupID: String) {
        if (party?.id == groupID) {
            party = null
        }

        guilds?.removeAll {
            it.id == groupID
        }

        parties?.removeAll {
            it.id == groupID
        }
    }
}
